src/redux/reducers/cart.js

```diff
@@ -0,0 +1,28 @@
+import * as actionTypes from '../actionTypes';
+
+const initialState = { productIds: [], quantity: {}, totalCartItems: 0 };
```

Comment:
Make it an array of objects, instead 🤔
```
[
    {
        id: '',
        quantity: '',
    }
]
```

`totalCartItems` can be the length of this array, and one less thing to maintain manually. 

Also, I'd recommend adding the whole product item here, instead of just ID.